newm-shared/build.gradle.kts

```diff
@@ -0,0 +1,76 @@
+plugins {
+    `java-library`
+    id(Dependencies.VersionsPlugin.ID)
+    id(Dependencies.ShadowPlugin.ID)
+    id(Dependencies.KtlintPlugin.ID)
+    kotlin(Dependencies.KotlinPlugin.JVM_ID)
+    kotlin(Dependencies.KotlinPlugin.SERIALIZATION_ID)
+}
+
+java.sourceCompatibility = JavaVersion.VERSION_17
+java.targetCompatibility = JavaVersion.VERSION_17
+
+ktlint {
+    version.set("0.43.2")
+}
+
+dependencies {
+    compileOnly(Dependencies.Kotlin.REFLECTION)
+    implementation(Dependencies.Kotlin.STDLIB_JDK8)
+
+    implementation(Dependencies.Coroutines.CORE)
+    implementation(Dependencies.Coroutines.JDK8)
+
+    implementation(Dependencies.KotlinXSerialization.JSON)
+
+    implementation(Dependencies.Ktor.SERVER_CORE)
+    implementation(Dependencies.Ktor.SERVER_CIO)
+    implementation(Dependencies.Ktor.SERVER_CONTENT_NEGOTIATION)
```

Comment:
There's a lot of dependencies here that aren't needed in newm-chain. Are we sure they are all required by this shared library?

Reply:
I'd like the shared library to pull in a minimum set of dependencies that it needs.

Reply:
Good call. I updated the file to the minimum dependencies needed atm. We can always add more later as needed, rather than adding all now.